obspy/io/sac/sactrace.py

```diff
@@ -1158,10 +1176,23 @@ def read(cls, source, headonly=False, ascii=False, byteorder=None,
                                             checksize=checksize)
         if not debug_strings:
             for i, val in enumerate(hs):
-                val = _ut._clean_str(val, strip_whitespace=False)
+                if encoding.lower() != 'ascii':  # throw warning
+                    msg = ("Using encoding other than ASCII: {}\n"
+                           "All non-ASCII characters will be replaced by "
+                           "?".format(encoding))
```

Comment:
I was trying to guide the user using those warnings. But, if we decided to document 'encoding' flag in the higher level function `_internal_read_sac`, there is truly no need to warn a conscious user about the obvious thing. I will change it along with the docstring.